templates/api.mustache

```diff
@@ -0,0 +1,361 @@
+        ...
+
+{{/asyncio}}
+{{#vendorExtensions}}
+{{#x-equinix-metal-paginated-property}}
+    @validate_arguments
+    def {{operationId}}_all_pages(self, {{#allParams}}{{^vendorExtensions.x-equinix-metal-page-param}}{{paramName}} : {{{vendorExtensions.x-py-typing}}}{{^required}} = None{{/required}}, {{/vendorExtensions.x-equinix-metal-page-param}}{{/allParams}}{{#asyncio}}async_req: Optional[bool]=None, {{/asyncio}}**kwargs) -> {{#asyncio}}Union[{{{returnType}}}{{^returnType}}None{{/returnType}}, Awaitable[{{{returnType}}}{{^returnType}}None{{/returnType}}]]{{/asyncio}}{{^asyncio}}{{{returnType}}}{{^returnType}}None{{/returnType}}{{/asyncio}}:  # noqa: E501
+        """{{#isDeprecated}}(Deprecated) {{/isDeprecated}}{{{summary}}}{{^summary}}{{operationId}}{{/summary}}  # noqa: E501
+
+        This method is the same as {{operationId}}, but fetches resources from all the pages.
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please pass async_req=True
+
+        >>> thread = api.{{operationId}}({{#allParams}}}{{^vendorExtensions.x-equinix-metal-page-param}}{{paramName}}, {{/vendorExtensions.x-equinix-metal-page-param}}, {{/allParams}}async_req=True)
+        >>> result = thread.get()
+
+{{#allParams}} {{^vendorExtensions.x-equinix-metal-page-param}}
+        :param {{paramName}}:{{#description}} {{{.}}}{{/description}}{{#required}} (required){{/required}}{{#optional}}(optional){{/optional}}
+        :type {{paramName}}: {{dataType}}{{#optional}}, optional{{/optional}}
+{{/vendorExtensions.x-equinix-metal-page-param}}{{/allParams}}
+        :param async_req: Whether to execute the request asynchronously.
+        :type async_req: bool, optional
+        :param _request_timeout: timeout setting for a single page request.
+                                 If one number provided, it will be single 
+                                 page request timeout. It can also be a pair
+                                 (tuple) of (connection, read) timeouts.
+        :return: Returns the result object.
+                 If the method is called asynchronously,
+                 returns the request thread.
+        :rtype: {{returnType}}{{^returnType}}None{{/returnType}}
+
+        """
+        all_pages = None
+
+        page = 1
+        while True:
+            # Template comment: here we set var "page" to page number and then
+            # we call the base method with all the parameters listed (including
+            # page).
+            page_response = self.{{operationId}}({{#allParams}}{{paramName}}, {{/allParams}}**kwargs)  # noqa: E501
+            if all_pages is None:
+                all_pages = page_response
+            else:
+                all_pages.{{.}}.extend(page_response.{{.}})
```

Comment:
I don't see this as something to address now, but we may want to change the meta properties of all_pages so that they reflect a complete pull with no "next" page and an accurate record count. Off the top of my head, I don't know what a good value for  meta.per_page would be. It could be the complete record count, or it could be the initial per_page. If it is the initial per_page value, I could see some code wanting to compare the per_page to total records and thinking there are more records to fetch. 

In the future we may also want to reconsider how we handle fetch errors while looping.

Reply:
Re: meta properties in all_pages, in the metal-go version of this feature, we create a new `DeviceList` and collect the devices into that, so the object we return in that SDK has the zero-value for the meta properties.  @displague does that implementation resolve or merely change your concerns?  @t0mk is it possible to do something similar here?

Reply:
> does that implementation resolve or merely change your concerns

It's not behavior we need to define so long as we say it is undefined. Or we can just pick one. We should document whatever it does either way.

Between the two choices, a synthetic meta or a nil meta, I think the synthetic approach is less likely to trip someone up, the obvious use-cases would be handled (pages=1, per_page=total count, page=1, count=total count, next href = nil, etc).

Reply:
I don't believe we need to address this in this PR.